### PR TITLE
Bjorncs/certificate builder

### PR DESCRIPTION
**Pull request by bjorncs**
FYI @mpolden

### CODE REVIEW

**Review by freva — APPROVED** [review]

**Comment by bjorncs** [comment]
@freva PTAL

File: vespa-athenz/src/main/java/com/yahoo/vespa/athenz/tls/Pkcs10Csr.java [hunk 1]
```diff
@@ -27,4 +39,52 @@ PKCS10CertificationRequest getBcCsr() {
...
+                .collect(toList());
+    }
+
+    /**
+     * @return true if CA certificate
+     */
+    public boolean getBasicConstraints() {
+        return getExtensions()
+                .map(BasicConstraints::fromExtensions)
+                .findAny()
```
Comment (freva) [inline vespa-athenz/src/main/java/com/yahoo/vespa/athenz/tls/Pkcs10Csr.java]: How/why does this work? Does `getExtension()` always return 1 element or does `isCA()` return true for all of them?

**Review by freva — APPROVED** [review]

File: vespa-athenz/src/main/java/com/yahoo/vespa/athenz/tls/Pkcs10Csr.java (hunk 1 above)
Comment (bjorncs) [reply vespa-athenz/src/main/java/com/yahoo/vespa/athenz/tls/Pkcs10Csr.java]: I found an answer here: https://access.redhat.com/documentation/en-US/Red_Hat_Certificate_System/8.0/html/Admin_Guide/Standard_X.509_v3_Certificate_Extensions.html
> ... , certificate extensions appear in sequence and only one instance of a particular extension may appear per certificate; for example, a certificate may contain only one subject key identifier extension.

I have no idea why the BC API allows multiple instances of the same extension...

Comment (freva) [reply vespa-athenz/src/main/java/com/yahoo/vespa/athenz/tls/Pkcs10Csr.java]: Shouldn't `getExtensions()` return `Extensions` or `Optional<Extensions>`?

Or at least consider
```
    public boolean getBasicConstraints() {
        return getExtensions()
                .map(BasicConstraints::fromExtensions)
                .anyMatch(BasicConstraints::isCA);
    }
```

Comment (bjorncs) [reply vespa-athenz/src/main/java/com/yahoo/vespa/athenz/tls/Pkcs10Csr.java]: Good point. It should be `OptionaBoolean`.

**Comment by bjorncs** [comment]
@freva PTAL

**Review by freva — APPROVED** [review]
Nice :1st_place_medal:

**Pull request merged**